### PR TITLE
apply tolerance to floating point comparisons

### LINKED ISSUE #23: Add tolerance to floating-point comparisons in tests
Add an `almost_equals()` utility function in each language to apply a tolerance to floating-point comparisons in all tests.

### PR DESCRIPTION
**Pull request by mattldawson**
closes #23 

also, fixed some bugs in the Fortran interpolator. I was accidentally running the C interpolator tests twice, which probably isn't the best way to test the Fortran interpolator =)

### CODE REVIEW

File: include/aero/aero.h.in [hunk 1]
```diff
@@ -6,11 +6,35 @@
...
+/// If the absolute relative difference between the numbers is within a
+/// tolerance based on the precision of \c aero_real_t, the function returns
+/// true. Otherwise, it returns false.
+static bool almost_equal( const aero_real_t a, const aero_real_t b ) {
+  if (a == 0.0 && b == 0.0) return true;
+  if (fabs(a - b) * 2.0 / fabs(a + b) < 5.0 * aero_epsilon ) return true;
+  return false;
+}
+
+/// Compares floating-point numbers
```
Comment (jeff-cohere) [inline include/aero/aero.h.in:30]: Nitpick: I am not a believer in Doxygen's "brief" and "long" summaries. I get the idea, but in practice no one wants to write two versions of an explanation. It's hard enough getting folks to write a single one. :-)

File: tests/interpolator/CMakeLists.txt [hunk 2]
```diff
@@ -6,10 +6,10 @@ include(test_util)
 ################################################################################
 # Tests for the Interpolator class
 
-create_standard_test(NAME fortran_interpolator SOURCES interpolator.c)
```
Comment (jeff-cohere) [inline tests/interpolator/CMakeLists.txt:9]: Just a question I probably should've asked earlier: what is a "standard test"?

**Review by jeff-cohere — APPROVED** [review]
Looks good. Just a few minor comments. We may have to generalize the interpolator a bit to deal with GPU-ified arrays, since at the moment we're assuming that all data is stored in a location accessible by the CPU. I suppose we get to decide whether `data()` returns a CPU-sensible array in that case, or whether we make the interpolator do the required copying itself.

File: include/aero/aero.h.in (hunk 1 above)
Comment (mattldawson) [reply include/aero/aero.h.in:30]: I don't always use it, but I think for things like this it's good to give the details of how to use the function. Most of the time though, I just use the brief description if it's clear enough

File: tests/interpolator/CMakeLists.txt (hunk 2 above)
Comment (mattldawson) [reply tests/interpolator/CMakeLists.txt:9]: Just a test that links to the `aero` library and has the usual include directories. In cases where we want to use a mock version of some class, we might want to compile differently and then just use the `add_aero_test()` function.

**Comment by mattldawson** [comment]
It would be nice to not rely on the `data()` function too much (or maybe at all?), but we might need to build out more `array` functionality

**Pull request merged**